chain/chain/src/chain.rs

```diff
@@ -1546,10 +1546,21 @@ impl Chain {
         let block_height = block.height();
         let prev_block_hash = *block.prev_block_hash();
         let prev_block = self.get_block(&prev_block_hash)?;
+        let prev_prev_hash = prev_block.header().prev_hash();
         let prev_chunk_headers =
             Chain::get_prev_chunk_headers(self.epoch_manager.as_ref(), &prev_block)?;
-
         let epoch_id = self.epoch_manager.get_epoch_id_from_prev_block(&prev_block_hash)?;
+
+        let (is_caught_up, _) = self.get_catchup_and_state_sync_infos(
```

Comment:
So basically, what you meant is that `_` aka `Option<StateSyncInfo>` might be invalid when using `optimistic_block_hash`? And this part of the return value should never be used?

Reply:
If so, maybe a comment would be helpful

Reply:
Found a smarter way to do the hack - only return StateSyncInfo if Some(block_hash) is provided.